pydatalab/tests/server/test_blocks.py

```diff
@@ -0,0 +1,180 @@
+import pytest
+
+from pydatalab.apps import BLOCK_TYPES, BLOCKS
+
+
+@pytest.mark.dependency()
+def test_get_all_available_block_types():
+    """Test that we can enumerate all available block types."""
+    assert len(BLOCKS) > 0
+    assert len(BLOCK_TYPES) > 0
+
+    for block_class in BLOCKS:
+        assert hasattr(block_class, "blocktype")
+        assert hasattr(block_class, "name")
+        assert hasattr(block_class, "description")
+        assert block_class.blocktype in BLOCK_TYPES
+        assert BLOCK_TYPES[block_class.blocktype] == block_class
+
+
+@pytest.mark.dependency(depends=["test_get_all_available_block_types"])
+@pytest.mark.parametrize("block_type", list(BLOCK_TYPES.keys()))
+def test_create_sample_with_each_block_type(admin_client, block_type, default_sample_dict):
+    """Test creating a sample and adding each available block type via API."""
+    sample_id = f"test_sample_with_{block_type.replace('-', '_')}"
+    sample_data = default_sample_dict.copy()
+    sample_data["item_id"] = sample_id
```

Comment:
I wonder if this test could be expanded to use actual files out of the box, we can make a "fat" default sample that has a load of example data uploaded to it, then try to make the blocks with real files. Could be pushed to another PR, as you prefer!

Reply:
#1247